File: 02Segundo/Programacion_Diseno_Orientada_Objetos_PDOO/Napakalaki/src/napakalaki/DeathBadConsequence.java
package napakalaki;

/**
 * 
 * @author Carlos de la Torre
 */
public class DeathBadConsequence extends NumericBadConsequence {

    /**
     * Constructor con parámetros, inicializa los valores del objeto con
     * parámetros de entrada:
     *
     * @param text numero entero que indica cantidad de tesoros
     * @param death indica si el mal rollo conlleva muerte.
     */
    public DeathBadConsequence(String text, boolean death) {
        super(text, Player.NIVEL_MAXIMO, death, MAXTREASURES, MAXTREASURES);
    }
}

File: 02Segundo/Programacion_Diseno_Orientada_Objetos_PDOO/Rupakalaki/lib/napakalaki/death_bad_consequence.rb
module NapakalakiGame
  
  class DeathBadConsequence < NumericBadConsequence
    
    def initialize(text="",death=false)
      super(text,Player.MAX_LEVEL,death,BadConsequence.MAXTREASURES,BadConsequence.MAXTREASURES)
    end
    
    public_class_method :new
  end
end
